community/server/src/test/java/org/neo4j/server/rest/DiscoveryServiceIT.java

```diff
@@ -83,14 +83,39 @@ public void shouldRedirectOnHtmlRequest() throws Exception
         Client nonRedirectingClient = Client.create();
         nonRedirectingClient.setFollowRedirects( false );
 
-        JaxRsResponse clientResponse = new RestRequest(null,nonRedirectingClient).get(server().baseUri().toString(),MediaType.TEXT_HTML_TYPE);
+        JaxRsResponse clientResponse =
+                new RestRequest( null, nonRedirectingClient ).get( server().baseUri().toString(), TEXT_HTML_TYPE );
 
         assertEquals( 303, clientResponse.getStatus() );
     }
 
-    private JaxRsResponse getDiscoveryDocument() throws Exception
+    /**
+     * This test does not allow for the fact that the Bolt address could be bound to a totally different NIC to the
+     * HTTP address. It's a consequence of the decision to create the Bolt server as a Kernel extension rather than
+     * as a container to host the database.
+     * <p>
+     * What we really need *is* a container and from there we could have a single point of origin for Websockets.
+     * For now this test asserts the 80% case that Bolt and HTTP addresses come from the same host/ip. Whether this
+     * is true in any given production environment we cannot know for sure. So it's a band-aid, not a cure.
+     */
+    @Test
+    public void boltAddressShouldAppearToComeFromTheSameOriginAsTheHttpAddressEvenThoughThisIsMorallyHazardous()
+            throws Exception
     {
-        return new RestRequest(server().baseUri()).get();
+        // given
+        RestRequest request = new RestRequest( server().baseUri() ).host( "boltprotocol.org" );
+
+        // when
+        JaxRsResponse response = request.get();
+
+        // then
+        Map<String,Object> map = JsonHelper.jsonToMap( response.getEntity() );
+        assertThat( String.valueOf( map.get( "bolt" ) ),
+                containsString( "bolt://" + request.getHeaders().get("host") ) );
```

Comment:
Yes I'll amend it to also check for the port. The main concern was to test that the host part of the address was correct.